Tools/GNUMake/packages/Make.hdf5

```diff
@@ -0,0 +1,17 @@
+CPPFLAGS += -DAMREX_USE_HDF5
+
+ifndef AMREX_HDF5_HOME
+ifdef HDF5_DIR
+  AMREX_HDF5_HOME = $(HDF5_DIR)
+endif
+ifdef HDF5_HOME
+  AMREX_HDF5_HOME = $(HDF5_HOME)
+endif
+endif
+
+ifdef AMREX_HDF5_HOME
+  HDF5_ABSPATH = $(abspath $(AMREX_HDF5_HOME))
+  INCLUDE_LOCATIONS += $(HDF5_ABSPATH)/include
+  LIBRARY_LOCATIONS += $(HDF5_ABSPATH)/lib
+  LIBRARIES += -lhdf5 -lz -ldl -L$(HDF5_ABSPATH)/lib
```

Comment:
Even when `AMREX_HDF5_HOME` is not defined, we should have those `-l`s.

Reply:
you mean these three: `-lhdf5 -lz -ldl`?

Reply:
yes, they might be found without -L.

Reply:
And we probably also want to pass `-Xlinker=-rpath -Xlinker=$(HDF5_ABSPATH)/lib`.